zerver/views/realm.py

```diff
@@ -158,6 +167,19 @@ def update_realm(
     if invite_to_realm_policy is not None and not user_profile.is_realm_owner:
         raise OrganizationOwnerRequired()
 
+    if is_demo_organization is not None and not is_demo_organization:
+        if not user_profile.is_realm_owner:
+            raise OrganizationOwnerRequired()
+
+        if realm.demo_organization_scheduled_deletion_date is None:
+            raise JsonableError(_("Organization to be converted must be a demo organization."))
```

Comment:
I think "Only demo organizations can use this endpoint" is probably simpler copy.